document/workflow/variable-tree/data-source/skeleton-tree-data-source.ts

```diff
@@ -143,14 +143,19 @@ export class SkeletonTreeNode implements GraphicallyRepresented<Element>, Dragga
     }
 
 
-    markRecursively() {
-        this.marked = true;
-        this.children.forEach(child => child.markRecursively());
+    markRecursively(mark = true) {
+        this.marked = mark;
+        this.children.forEach(child => child.markRecursively(mark));
+    }
+
+
+    unmarkRecursively() {
+        this.markRecursively(false);
     }
 
 
     get allChildrenMarked(): boolean {
-        return !this.children.some(child => !child.marked);
+        return this.children.length > 0 && !this.children.some(child => !child.marked);
```

Comment:
How can the information  propagate towards root?
Should the marked state change?
In my test is the a complex Type containing a complex object with allChildrenMarked = true not marked, but it should be.